pkg/assets/builder.go

```diff
@@ -127,32 +148,117 @@ func (a *AssetBuilder) RemapImage(image string) (string, error) {
 	}
 
 	a.ContainerAssets = append(a.ContainerAssets, asset)
-
 	return image, nil
 }
 
 // RemapFile sets a new url location for the file, if a AssetsLocation is defined.
-func (a AssetBuilder) RemapFile(file string) (string, error) {
+func (a *AssetBuilder) RemapFileAndSHA(file string, sha string) (string, *hashing.Hash, error) {
```

Comment:
Might be easier to build the FileAsset, with either a SHA or a SHAValue (or whatever we rename it to), and then call AssetBuilder to try to remap it.

Reply:
It is working ;) Also we are getting the sha values on the same pass, when we need to upload the cononical version. 

We get the original version during assets phase, upload it, and then check it during the other phases.